### PR TITLE
Conditional framing functions with cmake

### PR DESCRIPTION
**Pull request by Acuadros95**

### CODE REVIEW

File: src/c/profile/transport/custom/custom_transport.c [hunk 1]
```diff
@@ -24,13 +27,15 @@ static bool send_custom_msg(
     size_t bytes_written = 0;
     if (transport->framing)
     {
+        #ifdef UCLIENT_PROFILE_STREAM_FRAMING
```
Comment (pablogs9) [inline src/c/profile/transport/custom/custom_transport.c:30]:
```suggestion
#ifdef UCLIENT_PROFILE_STREAM_FRAMING
```

File: src/c/profile/transport/custom/custom_transport.c [hunk 2]
```diff
@@ -24,13 +27,15 @@ static bool send_custom_msg(
...
         bytes_written = uxr_write_framed_msg(&transport->framing_io,
                         (uxr_write_cb) transport->write,
                         transport,
                         buf,
                         len,
                         0x00,
                         &errcode);
+        #endif
```
Comment (pablogs9) [inline src/c/profile/transport/custom/custom_transport.c]:
```suggestion
#endif
```

File: src/c/profile/transport/custom/custom_transport.c [hunk 3]
```diff
@@ -69,6 +74,7 @@ static bool recv_custom_msg(
 
         if (transport->framing)
         {
+            #ifdef UCLIENT_PROFILE_STREAM_FRAMING
```
Comment (pablogs9) [inline src/c/profile/transport/custom/custom_transport.c:77]:
```suggestion
#ifdef UCLIENT_PROFILE_STREAM_FRAMING
```

File: src/c/profile/transport/custom/custom_transport.c [hunk 4]
```diff
@@ -77,6 +83,7 @@ static bool recv_custom_msg(
                             &remote_addr,
                             timeout,
                             &errcode);
+            #endif
```
Comment (pablogs9) [inline src/c/profile/transport/custom/custom_transport.c]:
```suggestion
#endif
```

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/410/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/424/)

File: src/c/profile/transport/custom/custom_transport.c [hunk 5]
```diff
@@ -146,11 +153,13 @@ bool uxr_init_custom_transport(
 
     if (transport->open(transport))
     {
+        #ifdef UCLIENT_PROFILE_STREAM_FRAMING
         if (transport->framing)
         {
             /* Init FramingIO. */
             uxr_init_framing_io(&transport->framing_io, 0x00);
         }
+        #endif
```
Comment (pablogs9) [inline src/c/profile/transport/custom/custom_transport.c]:
```suggestion
        if (transport->framing)
        {
#ifdef UCLIENT_PROFILE_STREAM_FRAMING
            /* Init FramingIO. */
            uxr_init_framing_io(&transport->framing_io, 0x00);
#else
            return false;
#endif
        }
```

**Review by pablogs9 — CHANGES_REQUESTED** [review]

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/414/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/428/)

File: src/c/profile/transport/custom/custom_transport.c [hunk 6]
```diff
@@ -1,8 +1,11 @@
 #include <uxr/client/profile/transport/custom/custom_transport.h>
 #include <uxr/client/profile/multithread/multithread.h>
-#include "../stream_framing/stream_framing_protocol.h"
 #include <uxr/client/util/time.h>
 
+#ifdef UCLIENT_PROFILE_STREAM_FRAMING
+#include "../stream_framing/stream_framing_protocol.h"
+#endif /* ifdef UCLIENT_PROFILE_STREAM_FRAMING */
```
Comment (pablogs9) [inline src/c/profile/transport/custom/custom_transport.c:7]:
```suggestion
#endif  // ifdef UCLIENT_PROFILE_STREAM_FRAMING
```

File: src/c/profile/transport/custom/custom_transport.c [hunk 7]
```diff
@@ -24,13 +27,15 @@ static bool send_custom_msg(
...
     {
+#ifdef UCLIENT_PROFILE_STREAM_FRAMING
         bytes_written = uxr_write_framed_msg(&transport->framing_io,
                         (uxr_write_cb) transport->write,
                         transport,
                         buf,
                         len,
                         0x00,
                         &errcode);
+#endif /* ifdef UCLIENT_PROFILE_STREAM_FRAMING */
```
Comment (pablogs9) [inline src/c/profile/transport/custom/custom_transport.c:38]:
```suggestion
#endif  // ifdef UCLIENT_PROFILE_STREAM_FRAMING
```

File: src/c/profile/transport/custom/custom_transport.c [hunk 8]
```diff
@@ -77,6 +83,7 @@ static bool recv_custom_msg(
                             &remote_addr,
                             timeout,
                             &errcode);
+#endif /* ifdef UCLIENT_PROFILE_STREAM_FRAMING */
```
Comment (pablogs9) [inline src/c/profile/transport/custom/custom_transport.c:86]:
```suggestion
#endif  // ifdef UCLIENT_PROFILE_STREAM_FRAMING
```

File: src/c/profile/transport/custom/custom_transport.c [hunk 9]
```diff
@@ -148,8 +155,12 @@ bool uxr_init_custom_transport(
     {
         if (transport->framing)
         {
+#ifdef UCLIENT_PROFILE_STREAM_FRAMING
             /* Init FramingIO. */
             uxr_init_framing_io(&transport->framing_io, 0x00);
+#else
+            return false;
+#endif /* ifdef UCLIENT_PROFILE_STREAM_FRAMING */
```
Comment (pablogs9) [inline src/c/profile/transport/custom/custom_transport.c:163]:
```suggestion
#endif  // ifdef UCLIENT_PROFILE_STREAM_FRAMING
```

**Review by pablogs9 — CHANGES_REQUESTED** [review]

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/415/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/429/)

**Review by pablogs9 — APPROVED** [review]

**Pull request merged**